flow/activities/flowable.go

```diff
@@ -186,7 +187,86 @@ func (a *FlowableActivity) CreateNormalizedTable(
 	return setupNormalizedTablesOutput, nil
 }
 
-func (a *FlowableActivity) StartFlow(ctx context.Context,
+func (a *FlowableActivity) MaintainPull(
+	ctx context.Context,
+	config *protos.FlowConnectionConfigs,
+	sessionID string,
+) error {
+	srcConn, err := connectors.GetCDCPullConnector(ctx, config.Source)
+	if err != nil {
+		return err
+	}
+	defer connectors.CloseConnector(srcConn)
+	a.CdcCacheRw.Lock()
+	a.CdcCache[sessionID] = srcConn
+	a.CdcCacheRw.Unlock()
+
+	ticker := time.NewTicker(15 * time.Second)
+	defer ticker.Stop()
+
+	/*
+			scan := make(chan struct{}, 1)
```

Comment:
This doesn't seem necessary, particularly full scan. Maybe move this to creating activity as a constant goroutine. Idea is that it'll detect connectors which have failed & remove them from map. ofc, then we need to handle in StartFlow when connector isn't in cache. This would make more sense then to get rid of MaintainPull having source connector tear down. But for drop purposes we'll want that anyway so we'd still need to remove here & there